src/steps/step.ts

```diff
@@ -0,0 +1,56 @@
+import { Project } from 'projen';
+import { JobStep } from 'projen/lib/github/workflows-model';
+import { Need } from 'projen/lib/gitlab';
+
+export interface GitlabStep {
```

Comment:
I dont like that we need different interfaces - shouldnt it rathe ber exactly one...?

Reply:
This is not something the user should ever see. This is the return type of the rendering. We need different types to render to to build Gitlab,Github, etc pipelines